client/settings/payment-method-settings/payment-request-button-preview.js

```diff
@@ -0,0 +1,103 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { React, useState, useMemo } from 'react';
+import { __, sprintf } from '@wordpress/i18n';
+import { Elements } from '@stripe/react-stripe-js';
+import { loadStripe } from '@stripe/stripe-js';
+
+/**
+ * Internal dependencies
+ */
+import PaymentRequestDemoButton from './payment-request-demo-button';
+import {
+	getPaymentRequestData,
+	shouldUseGooglePayBrand,
+} from 'payment-request/utils';
+import InlineNotice from 'components/inline-notice';
+
+/**
+ * stripePromise is used to pass into <Elements>'s stripe props.
+ * The stripe prop in <Elements> can't be change once passed in.
+ * Keeping this outside of <PaymentRequestButtonPreview> so that
+ * re-rendering does not change it.
+ */
+
+const PaymentRequestButtonPreview = ( props ) => {
+	const [ paymentRequest, setPaymentRequest ] = useState(
+		props.paymentRequest ?? null
+	);
+	const [ isLoading, setIsLoading ] = useState( props.isLoading ?? true );
```

Comment:
Looks like the props are being used only in the tests, and the state is also duplicated down in the `PaymentRequestDemoButton` component.

What do you think about removing the duplication of state in this component and down in the hierarchy, as well as removing these props used only for tests, so that the component's behavior in the tests is closer to how it would be used?

I was going to make a diff, but there are a few files that would need to be changed 😅 

Here's a comparison, tho, with the tests refactored: https://github.com/Automattic/woocommerce-payments/compare/fix/1808-add-digital-wallets-preview...frosso/digital-wallets-mocking
It also only uses `PaymentRequestButtonPreview` and gets rid of `PaymentRequestDemoButton`.

In that branch, I also included the suggestions I added in this PR (with the exclusion of the `key` prop one).

Reply:
Thanks for the suggestions.

> ..the state is also duplicated down in the PaymentRequestDemoButton component.

Can you expand on what "duplicate" means? My understanding is that passing down parent states to child components as props isn't considered to be duplicated. 

> What do you think about ... removing these props used only for tests, so that the component's behavior in the tests is closer to how it would be used?

👍  I had the `<PaymentRequestButtonPreview>` wrapping the Stripe's `<Elements>` because `Elements` is a dependency of the preview button, not a dependency of the `<PaymentRequestSettings>` component. I wanted to avoid loading the unnecessarily `<Elements>` component when the info is shown. 

In your branch, I like the `payment-request-button-preview.test.js` tests that mocks stripe, `shouldUseGooglePayBrand` to verify the 4 different previews. 

I can merge your branch in and I can update the `key` props for Safari.

Reply:
> Can you expand on what "duplicate" means? My understanding is that passing down parent states to child components as props isn't considered to be duplicated.

My apologies; I must have gotten confused while looking at the old `PaymentRequestButtonPreview` twice. You're right, `PaymentRequestButtonPreview` didn't have a duplicated `useState` within `PaymentRequestDemoButton` 👍 it was just that the props were being used only for tests.

> I had the `<PaymentRequestButtonPreview>` wrapping the Stripe's `<Elements>` because `Elements` is a dependency of the preview button, not a dependency of the `<PaymentRequestSettings>` component. I wanted to avoid loading the unnecessary `<Elements>` component when the info is shown. 

Ah, I see what you mean! Yeah, if we didn't want to have `Elements` being rendered in `PaymentRequestSettings`, a separate component would be necessary - and it can just render `Elements` with the stripe promise.
But `PaymentRequestSettings` doesn't seem too bad; it's now just rendering the `Elements` component with the `stripePromise` 😄